code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm

```diff
@@ -288,3 +317,266 @@
 	to_chat(X, "<span class='notice'>We reach out into mind of the creature, infecting their thoughts...</span>")
 	victim.hallucination += 100
 	add_cooldown()
+
+// ***************************************
+// *********** Sneak Stinger
+// ***************************************
+/datum/action/xeno_action/activable/sneak_stinger
+	name = "Stealth Stinger"
+	action_icon_state = "neuro_sting"
+	mechanics_text = "Use on an adjacent target while stealthed and your sneak attack is ready, while stationary or off run intent. Injects Acid while on Harm intent and Neurotoxin while on any other intent."
+	plasma_cost = HUNTER_STINGER_PLASMA_COST
+	keybind_signal = COMSIG_XENOABILITY_STEALTH_STINGER
+	cooldown_timer = HUNTER_STINGER_COOLDOWN
+
+/datum/action/xeno_action/activable/sneak_stinger/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return FALSE
+
+	if(QDELETED(A))
+		return FALSE
+
+	var/mob/living/carbon/xenomorph/hunter/X = owner
+	var/mob/living/carbon/human/C = A
+
+	if(!A?.can_sting())
+		to_chat(X, "<span class='xenowarning'>We cannot sting this target!</span>")
+		return FALSE
+
+	if(X.on_fire)
+		to_chat(X, "<span class='xenowarning'>We're too busy being on fire to sting them!</span>")
+		return FALSE
+
+	if(X.m_intent == MOVE_INTENT_RUN && ( X.last_move_intent > (world.time - HUNTER_SNEAK_ATTACK_RUN_DELAY) ) )//We can't sprint up to the target and use this.
+		to_chat(X, "<span class='xenowarning'>We must be stalking or stationary to properly sting the target!</span>")
+		return FALSE
+
+	if(!X.Adjacent(C))
+		to_chat(X, "<span class='xenowarning'>We must be adjacent to the target.</span>")
+		return FALSE
+
+	if(isnestedhost(C)) //no bully
+		to_chat(X, "<span class='xenowarning'>We refrain from unnecessarily bullying the host.</span>")
+		return FALSE
+
+	if(C.stat == DEAD)
+		to_chat(X, "<span class='xenowarning'>We care not for the deceased!</span>")
+		return FALSE
+
+	var/datum/action/xeno_action/stealth/S = locate() in X.xeno_abilities //Gotta reference the datum for whether we're stealthed/able to sneak attack
+
+	if(!S) //Sanity; should not be possible.
+		to_chat(X, "<span class='xenowarning'>We somehow don't have the ability to stealth!</span>")
+		return FALSE
```

Comment:
If it isnt possible add a CI check or a stacktrace here

Reply:
Done